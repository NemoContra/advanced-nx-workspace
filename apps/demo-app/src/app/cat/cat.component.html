<p>Only cats here</p>
<flight-workspace-shared-util-random-animal-image></flight-workspace-shared-util-random-animal-image>
<br />
<button flightWorkspaceSharedUiAnimalSoundButton="cat">Make sound</button>
