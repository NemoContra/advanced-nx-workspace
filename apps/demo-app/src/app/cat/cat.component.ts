import { Component } from '@angular/core';

@Component({
  selector: 'flight-workspace-cat',
  templateUrl: './cat.component.html',
  styleUrls: ['./cat.component.css'],
})
export class CatComponent {}
