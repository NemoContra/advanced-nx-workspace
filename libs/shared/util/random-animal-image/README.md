# shared-util-random-animal-image

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test shared-util-random-animal-image` to execute the unit tests.
