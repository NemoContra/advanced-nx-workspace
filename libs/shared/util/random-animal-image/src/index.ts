export * from './lib/shared-util-random-animal-image.module';
