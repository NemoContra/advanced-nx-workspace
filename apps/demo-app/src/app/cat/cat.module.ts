import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CatComponent } from './cat.component';
import { SharedUtilRandomAnimalImageModule } from '@flight-workspace/shared/util/random-animal-image';
import { RouterModule } from '@angular/router';
import { SharedUiAnimalSoundButtonModule } from '@flight-workspace/shared/ui/animal-sound-button';

@NgModule({
  declarations: [CatComponent],
  imports: [
    CommonModule,
    SharedUtilRandomAnimalImageModule.forChild({ cats: true, dogs: false }),
    RouterModule.forChild([
      {
        path: '',
        component: CatComponent,
      },
    ]),
    SharedUiAnimalSoundButtonModule,
  ],
  exports: [CatComponent],
})
export class CatModule {}
