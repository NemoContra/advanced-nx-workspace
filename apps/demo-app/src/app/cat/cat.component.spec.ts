import { createRoutingFactory } from '@ngneat/spectator/jest';
import { CatComponent } from './cat.component';
import { MockModule } from 'ng-mocks';
import { SharedUtilRandomAnimalImageModule } from '@flight-workspace/shared/util/random-animal-image';

describe('CatComponent', () => {
  const createComponent = createRoutingFactory({
    component: CatComponent,
    routes: [
      {
        path: '',
        component: CatComponent,
      },
    ],
    imports: [MockModule(SharedUtilRandomAnimalImageModule)],
  });

  it('should render', () => {
    const spectator = createComponent();
    expect(spectator.fixture).toMatchSnapshot();
  });
});
