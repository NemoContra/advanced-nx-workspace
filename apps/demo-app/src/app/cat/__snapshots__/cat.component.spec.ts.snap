// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatComponent should render 1`] = `
<flight-workspace-cat
  __ngContext__={[Function LRootView]}
>
  <p>
    Only cats here
  </p><flight-workspace-shared-util-random-animal-image /><br /><button
    flightworkspaceshareduianimalsoundbutton="cat"
  >
    Make sound
  </button>
</flight-workspace-cat>
`;
